cmake/onnxruntime_providers_cuda.cmake

```diff
@@ -122,17 +122,28 @@
   if (onnxruntime_REDUCED_OPS_BUILD)
     substitute_op_reduction_srcs(onnxruntime_providers_cuda_src)
   endif()
+
```

Comment:
do these changes need to be done under if(WIN32) ?

Reply:
It wasn't explicitly required as openvino and cpu didn't check and all CI passed too. But I agree that your suggestion is just cleaner. I don't feel right relying on this being ignored on non-Windows.

Addressed.